src/components/Popup/Popup.js

```diff
@@ -1,28 +1,63 @@
-import React from "react";
+import React, { useState } from "react";
 import { Popup } from "react-leaflet";
 import {
   Container,
   Button,
   PopupContainer,
   DeleteButton,
 } from "./Popup.styles";
+import WasTherePopup from "./WasTherePopup";
+
+const CustomPopup = ({ onDelete, onUpdate }) => {
+  const [showWasTherePopup, setShowWasTherePopup] = useState(false);
+
+  function handleWasThereClick(e) {
+    e.stopPropagation();
+    //onUpdate({ isVisited: true });
+    setShowWasTherePopup(true);
+  }
+
+  function handleWantThereClick(e) {
+    e.stopPropagation();
+    onUpdate({ isVisited: false });
+    //setShowWantTherePopup(true);
+  }
 
-const CustomPopup = ({ onWasThereClick, onWantThereClick, onDelete }) => {
   function handleDeleteClick(e) {
     e.stopPropagation();
+    onUpdate({ isVisited: false });
```

Comment:
Why are we updating and then deleting? I think (see previous comment) if you update the delete function so that it deletes from the database then there is no need to update anything first.